features/Extended Materials/Shaders/ExtendedMaterials/ExtendedMaterials.hlsli

```diff
@@ -5,6 +5,23 @@
 // http://www.diva-portal.org/smash/get/diva2:831762/FULLTEXT01.pdf
 // https://bartwronski.files.wordpress.com/2014/03/ac4_gdc.pdf
 
+struct DisplacementParams
+{
+	float DisplacementScale;
+	float DisplacementOffset;
+	float HeightScale;
+};
+
+float AdjustDisplacement(float displacement, DisplacementParams params)
+{
+	return (displacement - 0.5) * params.DisplacementScale + 0.5 + params.DisplacementOffset;
+}
+
+float4 AdjustDisplacement(float4 displacement, DisplacementParams params)
+{
+	return float4(AdjustDisplacement(displacement.x, params), AdjustDisplacement(displacement.y, params), AdjustDisplacement(displacement.z, params), AdjustDisplacement(displacement.w, params));
```

Comment:
Is this any better than simply copying `(displacement - 0.5) * params.DisplacementScale + 0.5 + params.DisplacementOffset;`?

Reply:
Hmmm no. No idea what I was thinking

Reply:
Oh, I remembered what I was thinking: that was because I needed the method for both float and float4 but didn't want to duplicate formula so that if it'd need to be changed there would be no need to change it in two places